cronitor/celery.py

```diff
@@ -0,0 +1,147 @@
+import typing
+import datetime
+import humanize
+import logging
+from cronitor import State, Monitor
+import cronitor
+import sys
+
+logger = logging.getLogger(__name__)
+try:
+    import celery
+    import celery.beat
+    from celery.schedules import crontab, schedule, solar
+    from celery.signals import beat_init, task_prerun, task_failure, task_success, task_retry
+
+    if typing.TYPE_CHECKING:
+        from typing import Dict, List, Union, Optional
+        import billiard.einfo
+        from celery.worker.request import Request
+except ImportError:
+    logger.error("Cannot use the cronitor.celery module without celery installed")
+    sys.exit(1)
+
+# For the signals to properly register, they need to be top-level objects.
+# Since they are defined dynamically in initialize(), we have to declare them up top,
+# make them global, and override them.
+celerybeat_startup = None
+ping_monitor_before_task = None
+ping_monitor_on_success = None
+ping_monitor_on_failure = None
+ping_monitor_on_retry = None
+
+
+def get_headers_from_task(task):  # type: (celery.Task) -> Dict
+    headers = task.request.headers or {}
+    headers.update(task.request.properties.get('application_headers', {}))
+    return headers
+
+
+def initialize(app, api_key=None):  # type: (celery.Celery, Optional[str]) -> None
+    if api_key:
+        cronitor.api_key = api_key
+
+    global celerybeat_startup
+    global ping_monitor_before_task
+    global ping_monitor_on_success
+    global ping_monitor_on_failure
+    global ping_monitor_on_retry
+
+    def celerybeat_startup(sender, **kwargs):  # type: (celery.beat.Service, Dict) -> None
+        scheduler = sender.get_scheduler()  # type: celery.beat.Scheduler
+        schedules = scheduler.get_schedule()
+        monitors = []  # type: List[Dict[str, str]]
+
+        for name in schedules:
+            if name.startswith('celery.'):
+                continue
+            entry = schedules[name]  # type: celery.beat.ScheduleEntry
+
+            # ignore all celerybeat scheduled events with the Cronitor exclusion header
+            headers = entry.options.pop('headers', {})
+            if headers.get('x-cronitor-exclude') in (True, 'true', 'True'):
```

Comment:
I love that you added a way to do this. How do you include the header when configuring a task?

Reply:
You can add headers when calling `apply_async()`.  There _may_ be a way to do it with the task decorator; I'm not sure. 

This definitely isn't the ideal end user API so I don't plan to leave it this way, which is why I haven't documented it yet

Reply:
Got it. Honestly it's not the worst either. Passing that header in the `apply_async` call on the rare occasion you want all but a certain task monitored is not bad.